### DISCUSSION TITLE
textobjetcs / treesitter

### DISCUSSION THREAD

**Discussion by comod** [JetBrains/ideavim#509] [Q&A]
Hi! Is there a better way to describe _motions_ to specific programming languages?
Lets say i want to to _operate_ something on a whole **php-class** or a whole **php-method**. Currently i must use `V%` on the first line of the class for example to only _select_ what i want.

Example:
```php
<?php

namespace Test;

class Foo {

    public function bar(): void
    {
        ...
    }

    // Cursor is here for example

}
```

`yac` - "yank as class" - would yank:

```php
class Foo {

    public function bar(): void
    {
        ...
    }
...
}
```

**Comment by AlexPl292** [comment]
Hi, unfortunately, that seems not possible now. I've found an issues for that in YouTrack: https://youtrack.jetbrains.com/issue/VIM-480/Add-text-objects-for-code-constructs-like-a-method-and-a-class
